draft-proxied-svcb-headers.md

```diff
@@ -17,132 +17,551 @@ author:
+(IPv4 or IPv6) that the proxy connected to for the CONNECT* request,
+or the value of a DNS CNAME on the path of aliases involved in
+resolving the hostname connected to by the proxy.
+
+The last member in the list MUST have an item value that is the string
+representation of the IPv6 or IPv4 address that represents the
+destination of the CONNECT* from the proxy.  If this is an IPv6
+address, it MUST follow the canonical string form from {{!RFC5952}}.
+
+*TODO*: Is there an IPv4 canonical form RFC we should reference here?
+
+If the hostname connected to by the proxy was a series of CNAMEs,
+these should be included at the front of the list, starting with
+the CNAME that the hostname resolved to and proceeding sequentially.
+
+Each item in the list has the following parameters:
+
+* t: The DNS RR type of the DNS query name, which MUST be specified
+  as an sf-integer (so 1 for A, 28 for AAAA, and 5 for CNAME).
+
+* ttl: The TTL of the address or CNAME record SHOULD be included as a
+  parameter with the key "ttl", whose value MUST be an sf-integer.
+  If the value of of intermediate records are unavailable,
+  the ttl parameter SHOULD be included on just the last list
+  entry, containing the minimum TTL value across the CNAMEs.
+
+* o: The owner name of the DNS record MAY be included.
+  This this parameter is specified, its value MUST be an sf-string
+  whose value is the owner name of the record.
+
+
+
+### Example Proxy-DNS-Used response
+
+As an example, assume that the proxy server resolved the
+authority hostname of "svc.example.com" as follows:
 
 ~~~ diagram
-   svc.example.com. 3600 IN HTTPS 1 svc2.example.com. alpn=h2,h3 echconfig="123..."
-   svc.example.com. 3600 IN HTTPS 2 . alpn=h2 echconfig="abc..."
+   svc.example.com.  7200 IN CNAME  svc.example.net.
+   svc.example.net.  1800 IN CNAME  svc2.example.net.
+   svc2.example.net.   60 IN A      192.0.2.74
+   svc2.example.net.   60 IN AAAA   2001:db8::75
+   svc2.example.net.   60 IN AAAA   2001:db8::76
 ~~~
 
-A successful CONNECT response would include the following headers, if the client requested both
-"alpn" and "echconfig":
+If the proxy connected to 2001:db8::75 then the successful CONNECT
+response would include the following header:
 
 ~~~ example
 HEADERS
 :method = CONNECT
 :status = 200
-dns-svcb-params = "svc2.example.com.";priority=1;ttl=3600;p1=:aDIsaDM=:;p5=:MTIzLi4u:,
-                  "svc.example.com.";priority=2;ttl=3600;p1=:aDI=:;p5=:YWJjLi4u:
+proxy-dns-used = "svc.example.net.";ttl=7200;t=5;o="svc.example.com.",
+                 "svc2.example.net.";ttl=1800;t=5;o="svc.example.net.",
+                 "2001:db8::75";ttl=60;t=28;o="svc2.example.net."
 ~~~
 
+
+*TODO*: Do we need the "o=" parameter or is it redundant?
+        If we include it, should it be a MUST or MAY?
+        
+*TODO*: Do we need to cover DNAME as well? 
+
+*TODO*: Should a future version be able to include NS record
+        and DNSSEC information?
```

Comment:
At first glance, it appears to be redundant. Is there a situation in which the owner is not simply the original hostname (for the first entry) or the preceding hostname (for subsequent entries)?

However, if we want to be able to include additional information in the future, it might make sense to swap the format around; rather than including a list of CNAME(s) and/or IP(s) and giving an owner for each, list the owner first and an inner list of useful records for each.

Reply:
The redundancy is why I'm include to leave it optional or just get rid of it.  It could be helpful for cases such as DNSSEC in the future, or perhaps if we include NS records or other things,